puppetdb/templates/puppetdb-deployment.yaml

```diff
@@ -49,20 +56,13 @@ spec:
             {{- toYaml .Values.puppetdb.init.jmx_exporter.resources | nindent 12 }}
       containers:
         - name: {{ .Chart.Name }}
+          securityContext:
+          {{- toYaml .Values.securityContext | nindent 12 }}
           image: "{{ .Values.puppetdb.image.repository }}:{{ .Values.puppetdb.image.tag }}"
           imagePullPolicy: {{ .Values.puppetdb.image.pullPolicy }}
-          command: [ "java" ]
-          args: [
-            "-Dlogappender=STDOUT", "-Djava.net.preferIPv4Stack=true",
-            "-Djdk.tls.ephemeralDHKeySize=2048",
-            "-Xms{{ .Values.puppetdb.java_mem_limit }}", "-Xmx{{ .Values.puppetdb.java_mem_limit }}", "-XX:+ExitOnOutOfMemoryError",
-            "-javaagent:/jmx-exporter/jmx_prometheus_javaagent-0.12.0.jar=9080:/jmx-exporter/config.yaml",
-            "-cp", "/opt/puppetlabs/server/apps/puppetdb/puppetdb.jar", "clojure.main",
-            "-m", "puppetlabs.puppetdb.cli.services",
-            "-c", "/etc/puppetlabs/puppetdb/conf.d/",
-            "--bootstrap-config", "/etc/puppetlabs/puppetdb/bootstrap.cfg",
-            "--restart-file", "/opt/puppetlabs/server/data/puppetdb/restartcounter"
-            ]
+          envFrom:
+          - secretRef:
+              name: {{ .Values.puppetdb.secretConfig }}
```

Comment:
I think you should document what kind of stuff can/should go in this secret.

What I don't really like about that is that if the secret does not exist the pods will fail.

Reply:
it is already documented in the values. 
Also, the non-existance of the secret making the pod fail (crash/not working) is a feature : k8s will warn the user that there is something missing.

Reply:
Maybe you should also let users put environment variable directly in the values ?

Setting non sensitive variables from a secret seem like a hassle, so having these directly in the values might help.

Reply:
In fact it would be far more flexible to allow complete freedom over the contents of `env` like this if the user wants to set environment variables from configmap or secret or another source they can.